src/Application/Commands/RedistributeMatchFunds.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Commands;
+
+use MatchBot\Domain\CampaignFundingRepository;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\FundingWithdrawal;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Redistribute match funding allocations where possible, from lower to higher priority match fund pots.
+ */
+class RedistributeMatchFunds extends LockingCommand
+{
+    protected static $defaultName = 'matchbot:redistribute-match-funds';
+
+    public function __construct(
+        private CampaignFundingRepository $campaignFundingRepository,
+        private DonationRepository $donationRepository
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription('Moves match funding allocations from lower to higher priority funds where possible');
+    }
+
+    protected function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        // TODO Change the fixed lookback to 2 days, or parameter-ise it, once CC23 is tidied in Prod.
+        $donationsToCheck = $this->donationRepository->findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(
+            new \DateTimeImmutable('-3 weeks')
```

Comment:
Using ambient context to get current time makes things hard to test - and I see here the test is only specifying that the argument should be of type DateTimeImmutable, not equal to any particular date.

Could we inject either a clock or a timestamp to make it more testable? Something like this, assuming the current is inject as a DateTimeImmutable `$this->now`

```suggestion
            $this->now->sub(new \DateInterval('P3W'))
```

Probably then the current date should come from the PSR11 container but I don't have a strong feeling about that vs it coming from `matchbot-cli.php`